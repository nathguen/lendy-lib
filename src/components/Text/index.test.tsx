import { render, screen } from '@testing-library/react-native';
import React from 'react';

import { Text, Wrapper } from '../';

describe('Text', () => {
  const text = 'Some text';

  it('renders Text', () => {
    const baseContainer = render(
      <Wrapper>
        <Text>{text}</Text>
      </Wrapper>
    );

    expect(baseContainer).toMatchSnapshot();
    expect(screen.getByText(text)).toBeTruthy();
  });

  it('renders Text with bold', () => {
    const boldContainer = render(
      <Wrapper>
        <Text bold>{text}</Text>
      </Wrapper>
    );

    expect(boldContainer).toMatchSnapshot();
    expect(screen.getByText(text)).toBeTruthy();
  });

  it('Text and Text with bold should not match', () => {
    const baseContainer = render(
      <Wrapper>
        <Text>{text}</Text>
      </Wrapper>
    );

    const boldContainer = render(
      <Wrapper>
        <Text bold>{text}</Text>
      </Wrapper>
    );

    expect(baseContainer.toJSON()).not.toEqual(boldContainer.toJSON());
  });
});
