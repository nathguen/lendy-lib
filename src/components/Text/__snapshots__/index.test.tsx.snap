// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders Text 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#191919",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
      }
    }
  >
    Some text
  </Text>
</View>
`;

exports[`Text renders Text with bold 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#191919",
        "fontSize": 17,
        "fontWeight": "700",
        "lineHeight": 22,
      }
    }
  >
    Some text
  </Text>
</View>
`;
